### PR TITLE
Update GKE versions after March 29 GKE release

### PR DESCRIPTION
**Pull request by pebrc**
https://cloud.google.com/kubernetes-engine/docs/release-notes#March_29_2021

### CODE REVIEW

File: .ci/pipelines/e2e-tests-gke-k8s-versions.Jenkinsfile [hunk 1]
```diff
@@ -35,7 +35,7 @@ pipeline {
         }
         stage('Run tests for different k8s versions in GKE') {
             parallel {
-                stage("1.15") {
+                stage("1.17") {
```
Comment (barkbay) [inline .ci/pipelines/e2e-tests-gke-k8s-versions.Jenkinsfile:38]: I think we should also update `runWith(lib, failedTests, '1.15', "eck-gke15-${BUILD_NUMBER}-e2e")`

File: .ci/pipelines/e2e-tests-gke-k8s-versions.Jenkinsfile [hunk 2]
```diff
@@ -46,7 +46,7 @@ pipeline {
                         }
                     }
                 }
-                stage("1.16") {
+                stage("1.18") {
```
Comment (barkbay) [inline .ci/pipelines/e2e-tests-gke-k8s-versions.Jenkinsfile:49]: Same here, I think we should update `runWith(lib, failedTests, '1.16', "eck-gke16-${BUILD_NUMBER}-e2e")`

File: .ci/pipelines/e2e-tests-gke-k8s-versions.Jenkinsfile (hunk 1 above)
Comment (pebrc) [reply .ci/pipelines/e2e-tests-gke-k8s-versions.Jenkinsfile:38]: 🤦 looks like I am still half asleep

**Review by barkbay — APPROVED** [review]
LGTM

**Comment by pebrc** [comment]
> failed to solve: rpc error: code = Unknown desc = executor failed running [/bin/sh -c go mod download]: exit code: 1

Jenkins test this please

**Comment by pebrc** [comment]
Jenkins test this please

**Pull request merged**